app.yaml

```diff
@@ -137,6 +137,74 @@ handlers:
   login: admin
   script: main_mail.app
   secure: always
+  
+# Serve static pages.
+- url: /splash?c=at1
+  static_files: core/templates/dev/head/dist/splash_at1.html
+  upload: core/templates/dev/head/dist/splash_at1.html
+  http_headers:
+    Cache-Control: 'no-cache, must-revalidate'
+    Expires: 'Mon, 01 Jan 1990 00:00:00 GMT'
```

Comment:
I think you can remove `Cache-Control` and `Expires`, because `expiration` sets it automatically, but please check it in your browser dev console.

Reply:
Yes they are set automatically, but not with the same values
`Cache-Control` is set to `public` while `Expires`  is set to the current date

Reply:
That's okay in this case. Since the pages are now static we can cache them for 0 seconds and save few KiBs by using cache validation.

Reply:
Done

Reply:
One question about this -- if we update the page, will the updates be displayed correctly to the user or will they get a cached version? I don't see any build hashes, so just wanted to double-check.

Reply:
I'll need to add the regex for that

Reply:
(To clarify, my previous question was more for @vojtechjelinek)

Reply:
Because the `Expires` will always be in the past, the browser will do a check if the file is the same as in the cache (by using hashes). If the file is different the new version will be sent, if the file is the same the cache version will be used.

Reply:
Ah ok, thanks! But that is the part I don't understand, since the URL is currently hardcoded. How would we use hashes in that case?

Also, I was just wondering -- why this is placed in the "dynamic" section of app.yaml when these are meant to be statically served?

Reply:
Everything will be handled by the HTTP. To simplify it: if the cache is not expired the cached version is used, if the cache is expired the HTTP makes an check call, if the file is the same 302 is returned if the file is different 200 is returned together with the new file.